modules/RTC/CodecSelection.js

```diff
@@ -98,50 +96,43 @@ export class CodecSelection {
-     * configured on the media session that is newly created.
-     *
-     * @param {JingleSessionPC} mediaSession media session that started.
-     * @returns {void}
-     * @private
-     */
-    _onMediaSessionStarted(mediaSession) {
-        const preferredCodec = mediaSession.isP2P ? this.p2pPreferredCodec : this.jvbPreferredCodec;
-        const disabledCodec = this.disabledCodec && this._isCodecSupported(this.disabledCodec)
-            ? this.disabledCodec
-            : null;
-
-        this._selectPreferredCodec(mediaSession, preferredCodec, disabledCodec);
-    }
-
-    /**
-     * Sets the codec on the media session based on the preferred codec setting and the supported codecs
+     * Sets the codec on the media session based on the preferred/disabled codec setting and the supported codecs
      * published by the remote participants in their presence.
      *
      * @param {JingleSessionPC} mediaSession session for which the codec selection has to be made.
-     * @param {CodecMimeType} preferredCodec preferred codec.
-     * @param {CodecMimeType} disabledCodec codec that needs to be disabled.
      */
-    _selectPreferredCodec(mediaSession = null, preferredCodec = null, disabledCodec = null) {
+    _selectPreferredCodec(mediaSession) {
         const session = mediaSession ? mediaSession : this.conference.jvbJingleSession;
-        const currentCodec = preferredCodec ? preferredCodec : this.jvbPreferredCodec;
-        let selectedCodec = currentCodec;
 
-        if (session && !session.isP2P && !this.options.enforcePreferredCodec) {
-            const remoteParticipants = this.conference.getParticipants().map(participant => participant.getId());
+        if (!session) {
+            return;
+        }
+        const preferredCodec = session.isP2P ? this.p2pPreferredCodec : this.jvbPreferredCodec;
+        const disabledCodec = session.isP2P ? this.p2pDisabledCodec : this.jvbDisabledCodec;
+        const currentCodec = session?.peerconnection.getConfiguredVideoCodec();
+        let selectedCodec = preferredCodec ?? currentCodec;
 
-            for (const remote of remoteParticipants) {
+        if (!this.enforcePreferredCodec) {
+            const remoteParticipants = this.conference.getParticipants().map(participant => participant.getId());
+            const remoteCodecs = remoteParticipants?.map(remote => {
                 const peerMediaInfo = session._signalingLayer.getPeerMediaInfo(remote, MediaType.VIDEO);
-                const peerCodec = peerMediaInfo?.codecType;
 
-                if (peerCodec
-                    && peerCodec !== currentCodec
-                    && (peerCodec !== CodecMimeType.VP9 || browser.supportsVP9())) {
-                    selectedCodec = peerCodec;
-                }
+                return peerMediaInfo?.codecType;
+            });
+
+            const nonPreferredCodecs = remoteCodecs.filter(codec => codec !== selectedCodec);
```

Comment:
Should we also filter out any disabled codecs?

Reply:
I am filtering them out in the next step but it makes sense to filter them out here. Will make the change.